### PR TITLE
Draft: Test/jest

### PR DESCRIPTION
**Pull request by xzd9326**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -91,7 +93,7 @@ module.exports = {
   // notifyMode: "failure-change",
 
   // A preset that is used as a base for Jest's configuration
-  // preset: undefined,
+  // preset: "ts-jest",
```
Comment (novaliu86) [inline jest.config.js]: neat: avoid unnecessary changes

File: test/tokenAsIf.spec.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  })
+
+
+  test("balanceOf", async () => {
+    provider = new JsonRpcProvider();
+    // calculate slotKey
+    const mappingSlotNumber = 9;
+    const ownerAddress = "0x1231DEB6f5749EF6cE6943a275A1D3E7486F4EaE";
+    const usdcTokenAddress = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+    const slotKey = getBalanceOfSlotKey(ownerAddress, mappingSlotNumber);
```
Comment (novaliu86) [inline test/tokenAsIf.spec.ts:50]: This two cases look good. Currently you still need to hardcode a valid `usdcTokenAddress` and its `mappingSlotNumber` here, because you don't have a good mechanism to mock the code to figure out `mappingSlotNumber`. If you do, you will be able to use any token addresses and `mappingSlotNumber`.

**Review by novaliu86 — APPROVED** [review]
Looks good to me. Feel free to check in after the clean up.

File: test/tokenAsIf.spec.ts (hunk 2 above)
Comment (xzd9326) [reply test/tokenAsIf.spec.ts:50]: Will try to resolve this in future versions.
